File: src/main/scala/br/com/caelum/livraria/techtalk/Livro.java
package br.com.caelum.livraria.techtalk;

import java.math.BigDecimal;

public class Livro {

    private String nome;
    private BigDecimal preco;

    public Livro(String nome, BigDecimal preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public BigDecimal getPreco() {
        return preco;
    }

    public void setPreco(BigDecimal preco) {
        this.preco = preco;
    }
}

File: src/main/scala/br/com/caelum/livraria/Livro.scala
package br.com.caelum.livraria

import java.lang.Long
import javax.persistence.{GeneratedValue, Id, Entity}
import org.hibernate.Session

@Entity
case class Livro(var nome:String, val preco:BigDecimal) {

  @Id @GeneratedValue
  var id:Long = _

}